schemes/sima_diagnostics/check_energy_diagnostics.F90

```diff
@@ -0,0 +1,86 @@
+! Diagnostic scheme for check_energy
+! Not all quantities are needed as diagnostics; this module is designed to ease development
+module check_energy_diagnostics
+   use ccpp_kinds, only: kind_phys
+
+   implicit none
+   private
+   save
+
+   public :: check_energy_diagnostics_init ! init routine
+   public :: check_energy_diagnostics_run  ! main routine
+
+CONTAINS
+
+   !> \section arg_table_check_energy_diagnostics_init  Argument Table
+   !! \htmlinclude check_energy_diagnostics_init.html
+   subroutine check_energy_diagnostics_init(errmsg, errflg)
+      use cam_history,         only: history_add_field
+      use cam_history_support, only: horiz_only
+
+      character(len=512), intent(out) :: errmsg
+      integer,            intent(out) :: errflg
+
+      ! Local variables:
+
+      errmsg = ''
+      errflg = 0
+
+      ! History add field calls
+      call history_add_field('cp_or_cv_dycore', 'specific_heat_for_air_used_in_dycore', 'lev', 'inst', 'J kg-1 K-1')
+      call history_add_field('scaling_dycore', 'ratio_of_specific_heat_for_air_used_in_physics_energy_formula_and_specific_heat_for_air_used_in_dycore_energy_formula', 'lev', 'inst', '1')
+
+      call history_add_field('te_cur_phys', 'vertically_integrated_total_energy_of_current_state_using_physics_energy_formula', horiz_only, 'inst', 'J m-2')
+      call history_add_field('te_cur_dyn', 'vertically_integrated_total_energy_of_current_state_using_dycore_energy_formula', horiz_only, 'inst', 'J m-2')
+      call history_add_field('tw_cur', 'vertically_integrated_water_vapor_and_condensed_water_of_current_state', horiz_only, 'inst', 'kg m-2')
```

Comment:
Thanks! I remember we went through a pretty lengthy discussion in https://github.com/ESCOMP/CAM/issues/1141#issuecomment-2354032103 about the `tw_` names and settled on `water_vapor_and_condensed_water` so I just wanted to confirm we want to make this change.

Reply:
In the Standard Name rules [here](https://github.com/ESCOMP/CCPPStandardNames/blob/main/StandardNamesRules.rst) it explicitly states the following:

 "_total_water_ can be used to designate water in every form, i.e. water vapor plus condensed water"
 
 So I think it's reasonable to use `total_water` here.  The discussion you are referring to in CAM was originally related to the use of `moist_air_and_condensed_water`, which we need for mixing ratios but do not want here because that also includes the mass of dry air, which isn't included in these quantities.
 
 Hopefully that all makes sense?  Also pinging @PeterHjortLauritzen here just to keep him in the loop.

Reply:
Thanks! I've updated throughout to use `vertically_integrated_total_water` for `tw_cur` (and `vertically_integrated_total_water_at_start_of_physics_timestep` for `tw_ini`)